src/coreclr/jit/emitarm64.cpp

```diff
@@ -16065,9 +16065,7 @@ bool emitter::IsRedundantMov(instruction ins, emitAttr size, regNumber dst, regN
 bool emitter::IsRedundantLdStr(
     instruction ins, regNumber reg1, regNumber reg2, ssize_t imm, emitAttr size, insFormat fmt)
 {
-    bool isFirstInstrInBlock = (emitCurIGinsCnt == 0) && ((emitCurIG->igFlags & IGF_EXTEND) == 0);
-
-    if (((ins != INS_ldr) && (ins != INS_str)) || (isFirstInstrInBlock) || (emitLastIns == nullptr))
```

Comment:
Why remove `emitLastIns == nullptr`?

Reply:
Hmm, looks like it was inadvertent. At any rate I will follow your suggestion to check this in the helper; I almost did that initially but was tempted to keep things a bit more minimal in case we wanted to port it back.